cmd/minishift/cmd/image/import.go

```diff
@@ -100,10 +100,14 @@ func importImage(cmd *cobra.Command, args []string) {
 		ImageMissStrategy: image.Skip,
 	}
 
-	err = handler.ImportImages(imageCacheConfig)
+	importedImages, err := handler.ImportImages(imageCacheConfig)
 	if err != nil {
 		atexit.ExitWithMessage(1, fmt.Sprintf("Container image import failed:\n%v", err))
 	}
+
+	if len(importedImages) < len(normalizedImageNames) {
+		atexit.ExitWithMessage(1, "At least one image could not be imported.")
```

Comment:
The problem is; which image failed? Why did it fail?

The last question, of why, will be hard as the error is not available. But the naming should be able to be determined as " A ∩ B ". Because I might have to re-export this one?

Reply:
> The problem is; which image failed? Why did it fail?

That's already displayed. I explicitly choose this more generic format for now.  I don't think any information is lost here. However, as you said, the information is there if one wants to extend it at some stage.

> Progress is shown, right? and the indication shows fail when an import fails... right?

Right

> If so, maybe it might be better to show: "Unable to continue since at least one image failed to import".

To continue with what. The commands end here.

Reply:
> Unable to continue

If the command exits, you have to express this to the user. Using "At least one image could not be imported." to me is not a clear exit condition by itself if expressed this way.

Reply:
> If the command exits, you have to express this to the user. Using "At least one image could not be imported." to me is not a clear exit condition by itself if expressed this way.

You have to look at all the messages which appear as part of running the command.

Reply:
@gbraad  I think @hferentschik is right here, check the output https://github.com/minishift/minishift/pull/1839#issuecomment-352414018 , though it is coming from start.go. But as per code it actually prints, ok, fail or cache miss against each image. So we are fine here. 

```
$ minishift images import --all
   Importing alpine:latest . FAIL
Container image import failed:
Error determining manifest MIME type for oci:/home/lmohanty/.minishift/cache/images:alpine:latest: open /home/lmohanty/.minishift/cache/images/blobs/sha256/c577c840a769fe3ecc2f7617640c03eca9ddad9b762208d3c9453fee8aa3d327: no such file or directory
```

Reply:
It was the message itself, but since the command actually 'continues' it can be ignored. But somehow this causes the instance to start with an incomplete set of images.

I do notice the alignments jump around here...
```
$ minishift images import --all
   Importing alpine:latest . FAIL
```
this is due to re-using the function for both the start.go logic and the cmd.